code/game/sound.dm

```diff
@@ -106,7 +106,12 @@ A good representation is: 'byond applies a volume reduction to the sound every X
 		S.falloff = falloff ? falloff : FALLOFF_SOUNDS * max(round(S.volume * 0.05), 1)
 
 	if(!is_global)
-		S.environment = SOUND_ENVIRONMENT_ROOM
+		//get area sound enviroment and use it for sound
+		var/area/A = get_area(src)
+		S.environment = A.sound_environment
+		//need room and roomHF to be not -10000 for reverb to work
+		S.echo[3] = 0
+		S.echo[4] = 0
```

Comment:
doubting myself one more time this is what tg does https://github.com/tgstation/tgstation/blob/186c9d8f7e4c9e7e1b7e8dd3dd778b9d232a9eb4/code/game/sound.dm#L167-L178

BYOND's SOUND_ENVIRONMENT_NONE does this, so that's why it had more reverb:
> The default value (-1) specifies no change in environment.

To get around this what tg does is keep the last environment used, but turn Room and RoomHF all the way down so nothing applies. So you would want this:
```suggestion
		if(S.environment != SOUND_ENVIRONMENT_NONE)
			//need room and roomHF to be not -10000 for reverb to work
			S.echo[3] = 0
			S.echo[4] = 0
```